test/unit/pipeline/test_train_model_sklearn.py

```diff
@@ -56,6 +61,25 @@ def test_split_data_train_validation_test_period(self):
         self.assertEqual(len(validation_data), 1296)
         self.assertEqual(len(test_data), 1)
 
+    def test_train_pipeline_happy(self):
+        """Test if happy flow of the train pipeline
+
+        The input data should not contain features (e.g. T-7d),
+        but it can/should include predictors (e.g. weather data)"""
+
+        # Run the this should return a model and report
+        with self.assertLogs() as captured:
+            model, report = train_model_pipeline(
+                pj=self.pj,
+                input_data=self.train_input,
+                # check_old_model_age=False,
+                # compare_to_old=False,
+            )
+        # Check if a new model was stored based on logging
+        assert captured.records[-1].getMessage() == "Fitted a new model, not yet stored"
```

Comment:
I think there should be a better way to check if a certain path through the code was taken. Comparing logger strings is a bit hacky and brittle.

Reply:
I did some searching, but this was the commonly proposed solution, at least without modifying the output of the original function

Reply:
Not sure where you found this solution, but In my opinion its very bad practise to do such a thing. Assuming you want to check with this unit test if the model was actually fitted I would propose the following solution. sklearn has a `check_is_fitted` function which will raise a `NotFittedError` if the model is not fitted.

```python
    def test_train_model_pipeline_happy_flow(self):
        """Test happy flow of the train model pipeline
            
            NOTE this does not explain WHY this is the case?
            The input data should not contain features (e.g. T-7d),
            but it can/should include predictors (e.g. weather data)

        """
        model, report = train_model_pipeline(pj=self.pj, input_data=self.train_input)

        # check if the model was fitted (raises NotFittedError when not fitted)
        self.assertIsNone(sklearn.utils.validation.check_is_fitted(model))

        # check if model is sklearn compatible
        self.assertTrue(isinstance(model, sklearn.base.BaseEstimator))

        # check if report is a Report
        self.assertTrue(isinstance(report, Report))
```

Reply:
Great, I'll use that.
Looking back at the code, I first made this unit test before your latest commits to the pipeline. Then, the pipeline did not return a model of report but stored them to disk, and had more paths that could be executed. For the current pipeline, your proposed tests makes a lot more sense, thanks.